docs/web/howto/client.rst

```diff
@@ -331,74 +331,85 @@ This function returns a ``Deferred`` that fires with the body after the request
+Be sure to ``pip install twisted[tls]`` in order to get all the dependencies necessary to do TLS properly, but beyond that you should not need to do much.
 
+However, for some uses, you may need to customize Agent's use of HTTPS; for example, to provide a client certificate, or to use a custom certificate authority for an internal network.
 
+Here, we're just going to show you how to inject the relevant TLS configuration into an Agent, so we'll use the simplest possible example, rather than a more useful but complex one.
 
+``Agent``'s constructor takes an optional second argument, which allows you to customize its behavior with respect to HTTPS.
+The object passed here must implement the :api:`twisted.web.iweb.IPolicyForHTTPS` interface.
 
-
-Here's an example which shows how to use ``Agent`` to request
-an *HTTPS* URL with no certificate verification.
-
-
-    
-
-
+So, here's our example.
+Using the very helpful ``badssl.com`` web API, we will construct a requst that fails to validate, because the certificate has the wrong hostname in it.
+The following Python program should produce a verification error, when run as ``python example.py https://wrong.host.badssl.com/``.
 
 .. code-block:: python
 
-    
-    from twisted.python.log import err
-    from twisted.web.client import Agent
-    from twisted.internet import reactor
-    from twisted.internet.ssl import optionsForClientTLS
-
-    def display(response):
-        print("Received response")
-        print(response)
-    
-    def main():
-        contextFactory = optionsForClientTLS(u"https://example.com/")
-        agent = Agent(reactor, contextFactory)
-        d = agent.request("GET", "https://example.com/")
-        d.addCallbacks(display, err)
-        d.addCallback(lambda ignored: reactor.stop())
-        reactor.run()
-    
-    if __name__ == "__main__":
-        main()
+    import sys
 
+    from twisted.internet.task import react
 
+    from twisted.web.client import Agent, ResponseFailed
 
-    
-
+    @react
+    def main(reactor):
+        agent = Agent(reactor)
+        requested = agent.request(b"GET", sys.argv[1])
+        def gotResponse(response):
+            print(response.code)
+        def noResponse(failure):
+            failure.trap(ResponseFailed)
+            print(failure.value.reasons[0].getTraceback())
+        return requested.addCallbacks(gotResponse, noResponse)
 
+This is because the certificate returned by ``wrong.host.badssl.com`` is actually for ``badssl.com``, and the hostnames don't match.
+In our pretend scenario, we want to construct an ``Agent`` that validates HTTPS certificates normally, _except_ for this one host, where it ignores the hostname mismatch and supplies the correct hostname manually, to work around this problem.
```

Comment:
Probably works better as two sentences with the split right around "where it ..." (and with some minor rewording).